### DISCUSSION TITLE
ProxyRoute - Match, Cluster - Destinations are read only in latest package

### DISCUSSION THREAD

**Discussion by npalla** [dotnet/yarp#732] [Q&A] [answered]
HI All,

I just started with using ReverseProxy, got package 1.0.0-preview.8.21065.1 and the property Match from ProxyRoute class is readonly and property Destinations from class Cluster is readonly. I am trying to use the Code example from repo and not able to create routes/clusters because of this. Is there any other way to do that?

![image](https://user-images.githubusercontent.com/3278465/107532716-2e38cd00-6b8c-11eb-91f0-9ae7720d040c.png)

![image](https://user-images.githubusercontent.com/3278465/107532802-44df2400-6b8c-11eb-9acf-b6238bc3a25e.png)

I believe the example samples given in the repo may not work in the latest package as they are recently checked in and the build is not yet published. Please suggest if there is any other workaround for now.

Thanks

**Comment by Tratcher** [comment, marked as answer]
The repo samples use the latest checked in version. For the released previews you should check the [docs](https://microsoft.github.io/reverse-proxy/articles/configproviders.html#example) or the samples from that release branch.
https://github.com/microsoft/reverse-proxy/blob/5e90b7808eb26cf88b8e0ca34d8b84afa66bc58e/samples/ReverseProxy.Code.Sample/Startup.cs#L35-L42

There was a breaking change here for preview 9 to use C# 9 record types. You can try that by downloading the daily builds from here:
https://github.com/microsoft/reverse-proxy/blob/main/docs/DailyBuilds.md

**Reply by npalla-tyler** [reply]
Thank you will do that.
